### PR TITLE
Refactor redirects edit view to CBV

### PR DESCRIPTION
**Pull request by rohitsrma**
In this PR I've converted redirects edit view from FBV to CBV as a part of https://github.com/wagtail/wagtail/issues/8365 .

### CODE REVIEW

File: wagtail/contrib/redirects/views.py [hunk 1]
```diff
@@ -127,49 +127,24 @@ def header_more_buttons(self):
...
-    if request.method == "POST":
-        form = RedirectForm(request.POST, request.FILES, instance=theredirect)
-        if form.is_valid():
-            with transaction.atomic():
-                form.save()
-                log(instance=theredirect, action="wagtail.edit")
-            messages.success(
-                request,
-                _("Redirect '%(redirect_title)s' updated.")
-                % {"redirect_title": theredirect.title},
```
Comment (laymonage) [inline wagtail/contrib/redirects/views.py:148]: This is one of those instances where I'm not really sure whether to remove the more specific string in favour of the generic one from the generic `EditView`. If we already have a more specific translatable string like this, I think it's good to keep it. It provides better context for translators compared to `"%(model_name)s '%(object)s' updated."`, which can be useful for languages with gendered nouns.

```suggestion
    def get_success_message(self):
        return _("Redirect '%(redirect_title)s' updated.") % {"redirect_title": self.object.title}
```

File: wagtail/contrib/redirects/views.py [hunk 2]
```diff
@@ -127,49 +127,24 @@ def header_more_buttons(self):
...
-                buttons=[
-                    messages.button(
-                        reverse("wagtailredirects:edit", args=(theredirect.id,)),
-                        _("Edit"),
-                    )
-                ],
-            )
-            return redirect("wagtailredirects:index")
-        else:
-            messages.error(request, _("The redirect could not be saved due to errors."))
```
Comment (laymonage) [inline wagtail/contrib/redirects/views.py:158]: Same here, we could set `error_message` on the view class instead:

```suggestion
    error_message = gettext_lazy("The redirect could not be saved due to errors.")
```

**Review by laymonage — COMMENTED** [review]
Thanks! This is a good start, but there are a few things missing and some that could be improved. Some of them are things we've discussed before, so I highly suggest re-reviewing your previous PRs to avoid the same mistakes – which can help us move faster 😄

**Comment by rohitsrma** [comment]
Thanks @laymonage, I've made the changes. And, I'm really sorry for making the same mistakes again. I'll make sure not to do them again in the future.

File: wagtail/contrib/redirects/views.py [hunk 3]
```diff
@@ -127,49 +127,21 @@ def header_more_buttons(self):
...
+class EditView(generic.EditView):
+    model = Redirect
+    form_class = RedirectForm
+    template_name = "wagtailredirects/edit.html"
+    index_url_name = "wagtailredirects:index"
+    edit_url_name = "wagtailredirects:edit"
+    delete_url_name = "wagtailredirects:delete"
+    pk_url_kwarg = "redirect_id"
+    permission_policy = permission_policy
+    error_message = gettext_lazy("The redirect could not be saved due to errors.")
```
Comment (laymonage) [inline wagtail/contrib/redirects/views.py:139]: We can now show the breadcrumbs

```suggestion
    error_message = gettext_lazy("The redirect could not be saved due to errors.")
    header_icon = "redirect"
    _show_breadcrumbs = True
```

**Review by laymonage — COMMENTED** [review]
One small thing remaining (enabling breadcrumbs), but otherwise this looks good to me! I'll add the breadcrumbs myself and merge. Thanks, and sorry if my previous comment was a bit harsh!

**Review by laymonage — APPROVED** [review]
Thanks! Looks like this also resolves an issue where the error message for duplicate redirects is rendered awkwardly on the page, so that's great!

## Before

image

## After

image

Note: this issue still exists in the create view though, but that's a separate task.

**Pull request merged**